### ISSUE TITLE
create vm with blob-url

### ISSUE THREAD

**Issue by mkliu** [Azure/azure-xplat-cli#1548]
tried to create vm using:
     azure vm create --custom-data=cloud-config.yaml --ssh=22 --ssh-cert=./myCert.pem --vm-size='extrasmall' --blob-url='https://xxx.blob.core.windows.net/vhds' --no-ssh-password --vm-name=node-1 --connect=xxx --location='West US' 2b171e93f07c4903bcad35bda10acf22__CoreOS-Stable-522.5.0 xxx 

It tells me 
     error:   Creating OS disks in $root storage container is not supported. Storage URL: https://housediary.blob.core.windows.net/$root/vhds
Version: 0.8.13

Basically I want to create the VHD in my specified container.

**Comment by jliu70** [comment]
I believe that you need to specify the VM disk name in blob-url

--blob-url='https://housediary.blob.core.windows.net/vhds/wayliu.vhd'

**Comment by kmouss** [comment]
use as this and let us know if it doesn't work:
azure vm create --custom-data=cloud-config.yaml --ssh=22 --ssh-cert=./myCert.pem --vm-size='extrasmall' --blob-url='https://xxx.blob.core.windows.net/vhds/some-blob-name.vhd' --no-ssh-password --vm-name=node-1 --connect=xxx --location='West US' 2b171e93f07c4903bcad35bda10acf22__CoreOS-Stable-522.5.0 xxx

**Comment by peterlimayo** [comment]
Same problem, found a fix:

create storage account xxx
create container yyy
then vm create using blob-url=https://xxx.blob.core.windows.net/yyy/some-name.vhd

otherwise, vm create put the root disk in the first storage account (alphabetical sort)
which may not be associated with the service that you spec'd as part of the cloudapp.net.
Nasty bug.

**Comment by kmouss** [comment]
Closing the issue, please feel free to re-open is you don't think the above suggestion works. thank you
